File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/ui/adapter/StopSearchAdapter.java
package com.tilly.steven.stlbusarrivals.ui.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.tilly.steven.stlbusarrivals.Model.Stop;
import com.tilly.steven.stlbusarrivals.R;

import java.util.ArrayList;

/**
 * Created by Steven on 2016-01-28.
 */
public class StopSearchAdapter extends ArrayAdapter<Stop> {

    private LayoutInflater inflater;
    private Context context;

    public StopSearchAdapter(Context context, int rowLayoutResourceId, ArrayList<Stop> list) {
        super(context, rowLayoutResourceId, list);
        this.inflater = LayoutInflater.from(context);
        this.context = context;
    }

    @SuppressLint("DefaultLocale")
    @Override
    public View getView(int position, View view, ViewGroup parent) {

        ViewHolder holder;
        if (view != null) {
            holder = (ViewHolder) view.getTag();
        } else {
            view = inflater.inflate(R.layout.row_stop_search_list, parent, false);
            holder = new ViewHolder();
            holder.stopName = (TextView) view.findViewById(R.id.tv_row_stop_name);
            view.setTag(holder);
        }

        Stop item = getItem(position);
        holder.stopName.setText(item.getTitle());

        return view;
    }

    static class ViewHolder {
        TextView stopName;
    }
}

File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/ui/adapter/StopSearchAdapter.kt
package com.tilly.steven.stlbusarrivals.ui.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.tilly.steven.stlbusarrivals.R
import com.tilly.steven.stlbusarrivals.model.Stop
import java.util.*

/**
 * Created by Steven on 2016-01-28.
 */
class StopSearchAdapter(ctx: Context, rowLayoutResourceId: Int, list: ArrayList<Stop>) : ArrayAdapter<Stop>(ctx, rowLayoutResourceId, list) {

    private val inflater: LayoutInflater = LayoutInflater.from(context)

    @SuppressLint("DefaultLocale")
    override fun getView(position: Int, view1: View?, parent: ViewGroup): View {
        var view = view1

        val holder: ViewHolder
        if (view != null) {
            holder = view.tag as ViewHolder
        } else {
            view = inflater.inflate(R.layout.row_stop_search_list, parent, false)!!
            holder = ViewHolder()
            holder.stopName = view.findViewById<View>(R.id.tv_row_stop_name) as TextView
            view.tag = holder
        }

        val item = getItem(position)
        holder.stopName.text = item?.title
        return view
    }

    internal class ViewHolder {
        lateinit var stopName: TextView
    }
}